core/src/collections.rs

```diff
@@ -56,16 +55,33 @@ lazy_static! {
     ];
 }
 
+#[derive(Debug, Clone)]
+pub(crate) struct MongoODBCCollectionSpecification {
+    pub name: String,
+    pub collection_type: CollectionType,
+}
+
+impl MongoODBCCollectionSpecification {
+    pub fn new(name: String, collection_type: CollectionType) -> Self {
+        Self {
+            name,
+            collection_type,
+        }
+    }
+}
+
 #[derive(Debug)]
 struct CollectionsForDb {
     database_name: String,
-    collection_list: Cursor<CollectionSpecification>,
+    collection_list: Vec<MongoODBCCollectionSpecification>,
 }
 
 #[derive(Debug)]
 pub struct MongoCollections {
     // The current collection specification.
-    current_collection: Option<CollectionSpecification>,
+    current_collection: Option<MongoODBCCollectionSpecification>,
+    // The index of the current collection.
+    current_collection_index: Option<usize>,
```

Comment:
I tried to match the style that was used in each file. Initially, both relied on Cursor methods to ensure proper iteration. In `collections`, there was a field meant to maintain an index into which database was being looked into, and the collections were being advanced with the Cursor. There I chose a vec and added an index variable to help iterate through the collections.

In `fields`, the databases are in a VecDeque are iterated using pop_front, and the collections were being iterated using a cursor method. I decided to use a VecDeque to match the style and avoid introducing another variable, especially since there wasn't an index tracking variable for the dbs.

I think changing it to -1 would add more code. I was trying to keep the changeset as small as possible. If I were to make larger changes, I'd probably try to remove any variables associated with index tracking and use an iterator or a VecDeque.